### PR TITLE
Test(eos_designs): Add test for duplicate detection

### LINKED ISSUE #2748: Test(eos_designs): Add negative unit tests for duplicate detection
So following discussion over chat here is the agreement to move forward:

The PR is merged as is and then two extra PRs should be opened:

1. to add negative tests for duplicate detection
2. to refactor all occurences of duplicate detection into a generic function in shared_utils and replace it

_Originally posted by @gmuloc in https://github.com/aristanetworks/ansible-avd/issues/2627#issuecomment-1517459908_

### PR DESCRIPTION
**Pull request by ghost**
## Change Summary

Adding unit tests in eos_designs_negative_unit_tests molecule scenario.

## Related Issue(s)

Fixes #2748 

## Component(s) name

`arista.avd.eos_designs`

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/group_vars/SPINES.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/group_vars/SPINES.yml]: Since there is only one switch leveraging these vars, just move it to that switch' hostvars file. Also change the hostname to follow the same style with duplicate....

File: ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/intended/structured_configs/duplicate-ip-address-1.yml
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/intended/structured_configs/duplicate-ip-address-1.yml:1]: I wonder if these files can be removed now, since the task would fail, so they are never generated. Probably just leftovers from your testing.

File: ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/duplicate-interfaces-in-point-to-point-services.yml
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/duplicate-interfaces-in-point-to-point-services.yml:1]: You have two files with almost the same name ("-in" being the only difference). Please call them -1 and -2 instead. Same for hostnames.

File: ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/duplicate-sub-interfaces-point-to-point-services.yml
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/duplicate-sub-interfaces-point-to-point-services.yml:1]: seems like a duplicate of ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/duplicate-interfaces-in-point-to-point-services.yml

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/duplicate-sub-interfaces-point-to-point-services.yml:1]: Looking at it again, it varies a bit, so maybe have them called -1 -2 -3

File: ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/intended/structured_configs/duplicate-ip-address-1.yml
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/intended/structured_configs/duplicate-ip-address-1.yml:1]: These files are getting generated on molecule run, as the error occurring by this configuration is for the uplink-switch i.e. `duplicate-ip-address-uplink-switch-router-bgp` [ Mentioned in the file duplicate-ip-address-uplink-switch-router-bgp.yml]

File: ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/group_vars/SPINES.yml (hunk 1 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/group_vars/SPINES.yml]: moved and changed the name to `duplicate-ip-address-uplink-switch-router-bgp`

File: ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/duplicate-interfaces-in-point-to-point-services.yml
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/duplicate-interfaces-in-point-to-point-services.yml:1]: Updated the file names for point-to-point-services

File: ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/duplicate-sub-interfaces-point-to-point-services.yml
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/duplicate-sub-interfaces-point-to-point-services.yml:1]: fixed

File: ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/duplicate-interfaces-point-to-point-services_1.yml
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/host_vars/duplicate-interfaces-point-to-point-services_1.yml:1]: minor detail. Use hyphen instead of underscore everywhere in hostnames.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM. Thanks!

**Comment by ClausHolbechArista** [comment]
@gmuloc let's get this one in _before_ #2766. Then I can adjust the tests as part of of #2766.

**Review by ClausHolbechArista — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM
